pkg/azuredisk/azure.go

```diff
@@ -20,26 +20,64 @@ import (
 	"fmt"
 	"os"
 
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog"
 	"k8s.io/legacy-cloud-providers/azure"
 )
 
 // GetCloudProvider get Azure Cloud Provider
-func GetCloudProvider() (*azure.Cloud, error) {
-	credFile, ok := os.LookupEnv("AZURE_CREDENTIAL_FILE")
-	if ok {
-		klog.V(2).Infof("AZURE_CREDENTIAL_FILE env var set as %v", credFile)
-	} else {
-		credFile = "/etc/kubernetes/azure.json"
-		klog.V(2).Infof("use default AZURE_CREDENTIAL_FILE env var: %v", credFile)
+func GetCloudProvider(kubeconfig string) (*azure.Cloud, error) {
+	klog.V(2).Infof("reading cloud config from secret")
+	kubeClient, err := getKubeClient(kubeconfig)
+	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
```

Comment:
`!os.IsNotExist(err) && err != rest.ErrNotInCluster`
The two judgments ensure that we are able to test the driver in a local machine.

Reply:
could you add a unit test for this? at least write a negative case for this functionality?

Reply:
done. Add some positive and negative cases.